### PR TITLE
Add support for swagger consistency checks

### PR DESCRIPTION
**Pull request by mastfish**
Usage: 
```ruby
be_consistent_with_swagger_at "example.com"
```
It will then then look at http://example.com/swagger.json, and check to make sure there are no inconsistencies.

### CODE REVIEW

File: lib/apivore/rspec_matchers.rb [hunk 1]
```diff
@@ -34,6 +34,25 @@ module RspecMatchers
       end
     end
 
+    matcher :be_consistent_with_swagger do |master_swagger_uri|
+
+      attr_reader :actual, :expected
+
+      match do |body|
+        req = Faraday.new(master_swagger_uri).get("swagger.json")
```
Comment (mastfish) [inline lib/apivore/rspec_matchers.rb]: This probably should use only the standard lib.

**Comment by hornc** [comment]
Can you provide a definition of 'inconsistencies'? Is it just that the definitions blocks in the two swagger.jsons should match?

**Comment by mastfish** [comment]
Kind of. The supplied uri should return a superset of all definitions, and the service being tested cannot have definitions which disagree with that source of truth.

So for example, at "example.com/swagger.json", there might be definitions:
product
category

And at the service being tested, there might be a definition:
category

This is fine, as long as the category documentation is identical between the two swaggers.

**Comment by leondewey** [comment]
@mastfish so if I get this right you want to enforce via apivore that we use /swagger.json ?

**Comment by mastfish** [comment]
Consider 2 services - one of them has a definition for "422Error" that refers to "product", and one has a definition of "422Error" that refers to "deal".

Then consider a third service, "api.westfield.io", that's trying to create a consistent swagger doc from both. Clearly, there's a merge conflict here, one that requires one (or both) of those services to change their "422Error" definitions.

This is designed so that trying to publish a "422Error" that disagrees with "api.westfield.io"'s swagger definitions throws an error, using it as a canonical source of truth.

It's opt-in, so upgrading apivore won't cause surprise errors, and it targets definitions only, because testing revealed that paths (should) have only one source of truth, so there's no need to cross-check them.

**Comment by chrisrbnelson** [comment]
LGTM! Will be nice to bring all the definitions in line with each other.

**Pull request merged**